0310-minimum-height-trees.py

```diff
@@ -0,0 +1,57 @@
+from collections import defaultdict, deque
+from typing import List
+
+class Solution:
+    def findMinHeightTrees(self, n: int, edges: List[List[int]]) -> List[int]:
+        if n <= 1:
+            return [0]  # Special case for when there is only one node or no nodes
+        
+        graph = defaultdict(list)
+        indegree = [0] * n
+        
+        for node1, node2 in edges:
```

Comment:
The variable names used in the code are short and concise, 
but they could be made more descriptive to enhance readability. 
For example, instead of using node1 and node2, you can use source and destination to better represent the edges.